addons/sale/models/sale_order.py

```diff
@@ -423,10 +423,9 @@ def _compute_team_id(self):
     def _compute_amounts(self):
         """Compute the total amounts of the SO."""
         for order in self:
-            order_lines = order.order_line.filtered(lambda x: not x.display_type)
-            order.amount_untaxed = sum(order_lines.mapped('price_subtotal'))
-            order.amount_total = sum(order_lines.mapped('price_total'))
-            order.amount_tax = sum(order_lines.mapped('price_tax'))
+            order.amount_untaxed = order.tax_totals['amount_untaxed']
+            order.amount_total = order.tax_totals['amount_total']
+            order.amount_tax = sum(tax_dict['tax_group_amount'] for groups in order.tax_totals['groups_by_subtotal'].values() for tax_dict in groups)
```

Comment:
Nope, do not base the computation of stored fields on a nightmary non stored field

Reply:
Would it be better to get the tax dict from `_prepare_tax_totals`? (If yes, we could merge `_compute_amounts` and `_compute_tax_totals`) 

By the way, `purchase` has the same code, so we need to patch it too

Reply:
@Feyensv 
After double checking the issue it seems the root cause is half in `sale`/`purchase`, half in `account`
As explained in commit message 1) we round the tax computed in `_compute_taxes`, regardless of settings and 2)  `order.amount_total` is a sum of already rounded values, we should sum `amount_untaxed` and `amount_tax`

Reply:
Ok for me, it's cleaner and it explains why the fix is only necessary in 16.0 (cf. cd6c33575b4823ebf6ab7100552c668360f6b30d)

Leaving the final validation to accounting team :ok_hand: